src/models/ServerAcl.ts

```diff
@@ -27,6 +28,25 @@ export class ServerAcl {
     private deniedServers: Set<string> = new Set<string>();
     private allowIps = false;
 
+    public constructor(public readonly homeserver: string) {
+
+    }
+
+    /**
+     * Checks the ACL for any entries that might ban ourself.
+     * @returns A list of deny entries that will not ban our own homeserver.
+     */
+    public safeDeniedServers(): string[] {
```

Comment:
Test looks good. I don't understand why we don't rather rather test whenever we add a new entry to `deniedServers`, though.

Reply:
It's a precaution if someone somehow mutates the deny list without using that method in the future somehow. It is also because the `literalAclContent` method still exists, though we don't use it in any of our own code.

Reply:
Makes sense. Could you mention that as a comment?